### PR TITLE
Adding a new node with searcher

### LINKED ISSUE #8064: Adding a new node with searcher
Pressing enter should open component browser. Accepting CB expression should add a new node with this expression. 

The CB position should be placed properly according to functions implemented in #8069

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Implements #8064 /

@kazcw Could you double-check that this plays well with your edge dragging implementation. Some of the code seemed a work in progress, and I want to avoid any issues with what you had planned there. 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor.vue [hunk 1]
```diff
@@ -169,22 +207,10 @@ function setCurrentInteraction(interaction: Interaction | undefined) {
 }
 
 function cancelCurrentInteraction() {
+  currentInteraction.value?.cancel()
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor.vue]: This is called in `setCurrentInteraction`. We could call it here and set `currentInteraction` directly (and then `setCurrentInteraction` no longer needs to be able to accept `undefined`), but it's important not to call it twice.

File: app/gui2/src/components/GraphEditor.vue [hunk 2]
```diff
@@ -256,7 +290,10 @@ watch(
     @drop.prevent="handleFileDrop($event)"
   >
     <svg :viewBox="navigator.viewBox">
-      <GraphEdges @startInteraction="setCurrentInteraction" @endInteraction="interactionEnded" />
+      <GraphEdges
+        @startInteraction="setCurrentInteraction"
+        @endInteraction="cancelCurrentInteraction"
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor.vue]: The function passed here will be the way the GraphEdges component can remove its Interaction from the currentInteraction when it completes. I don't think that in general we should assume that we can always run an interaction's cancellation code when ending the interaction.

**Review by kazcw — CHANGES_REQUESTED** [review]
Sorry, the Interaction API is lacking documentation and probably hard to make sense of without it. The idea is to expose operations applicable to any current user interaction. Currently, an interaction may capture clicks, an interaction may be cancelled, or an interaction may end in a way that doesn't run the cancellation code. On top of this interface is provided mutual exclusivity, so that e.g. if the user is dragging an edge and presses Enter the edge drag is cancelled.

Some interactions like edge-dragging cannot succeed without further information/user input. I don't see how we could ever know that the current interaction has succeeded/should succeed except in interaction-specific code. If success is interaction-specific, I don't think it should be exposed generically. Rather, the current design offers a way for a component providing an Interaction to remove itself from the `currentInteraction` in case an interaction-specific non-cancellation ending occurs--including successful completion of the interaction.

File: app/gui2/src/components/GraphEditor.vue [hunk 3]
```diff
@@ -155,12 +155,50 @@ const groupColors = computed(() => {
...
+class EdgeDragging extends Interaction {
+  nodeId: ExprId
+
+  constructor(nodeId: ExprId) {
+    super()
+    this.nodeId = nodeId
+  }
+
   cancel() {
     componentBrowserVisible.value = false
```
Comment (kazcw) [inline app/gui2/src/components/GraphEditor.vue:197]: Previously this was part of the `EditingNode` interaction. I don't see how closing the CB would be a normal part of cancelling an `EdgeDragging` operation.

File: app/gui2/src/components/GraphEditor.vue (hunk 1 above)
Comment (MichaelMauderer) [reply app/gui2/src/components/GraphEditor.vue]: I divided it up into three methods: `setCurrentInteraction` without undefined, `cancelCurrentInteraction` (which calls `cancel` and `abortCurrentInteraction`, which does not call `cancel`. I think this is the most explicit way to handle it. By default, `setCurrentInteraction` cancels the previous interaction, but if needed, `abortCurrentInteraction` can be called before setting a new one. Does that seem reasonable?

File: app/gui2/src/components/GraphEditor.vue (hunk 2 above)
Comment (MichaelMauderer) [reply app/gui2/src/components/GraphEditor.vue]: I exchanged it for the new `abortCurrentInteraction`.

File: app/gui2/src/components/GraphEditor.vue (hunk 3 above)
Comment (MichaelMauderer) [reply app/gui2/src/components/GraphEditor.vue:197]: Removed.

**Review by kazcw — APPROVED** [review]

**Review by Frizi — APPROVED** [review]

**Pull request merged**